Use flexbox on app page

rullzer opened the pull request:
The amount of white space on the app page was annoying me. So I decided to give flexbox a try.

Before:
![before](https://cloud.githubusercontent.com/assets/45821/18410392/280bfba0-7762-11e6-8551-f3a4f77f32ce.png)

After:
![after](https://cloud.githubusercontent.com/assets/45821/18410395/2d03e4ec-7762-11e6-8773-c9c098894086.png)

Which looks a lot better to me.

However it does need work as you can see on the notenabled page:
![not_enabled](https://cloud.githubusercontent.com/assets/45821/18410461/ead5e9a6-7763-11e6-851b-f97a03c60c4e.png)

Far from done.
- [x] Experimental apps display
- [ ] Fix per app layout a bit more
- [ ] Full Safari/IE support
- [ ] Restrict to apps only (currently affects personal/admin settings, too)

CC: @nextcloud/designers @MorrisJobke

rullzer committed:
Use flexbox on app page

raghunayyar commented:
Personally, I find it a bit too noisy now since the left ones have the heading on the left and the centre ones have the heading below the image. There is no visual consistency here. May be you want to use two of them instead of the three and make sure the image comes on the left and the content on the right.

raghunayyar commented:
Aaah, I missed the second image. Never mind. :P

tcitworld commented:
I think margin for only two columns would be fine.

eppfel committed:
Limit to apps list, add proper flex attributes

Let app-name headline break, increase padding on the right for more visual clearance.
Make experimental header full width

Signed-off-by: Felix A. Epp <work@felixepp.de>

eppfel commented:
I like the idea, but it could really be a bit too "noisy". The styling of the per app layout is not so easy, because these are inline-box elements right now.
Experimental apps are separated now:
<img width="1482" alt="bildschirmfoto 2016-09-10 um 17 06 16" src="https://cloud.githubusercontent.com/assets/1899308/18411511/fdb81d22-7778-11e6-9019-e6257091a3d4.png">
<img width="1152" alt="bildschirmfoto 2016-09-10 um 17 06 30" src="https://cloud.githubusercontent.com/assets/1899308/18411512/020b49a8-7779-11e6-8639-36453c183d88.png">

rullzer commented:
Thanks for the feedback. I just wanted to get things started. As I have the artistic insight of a table I'll happily hand this over :) I just found that the current design had way to much wasted space.

@eppfel thnx!

eppfel commented:
My try in making it less noisy:
<img width="2560" alt="bildschirmfoto 2016-09-11 um 11 49 33" src="https://cloud.githubusercontent.com/assets/1899308/18416556/2350fff0-7818-11e6-8497-4f5e2534dbe7.png">
<img width="1406" alt="bildschirmfoto 2016-09-11 um 11 51 57" src="https://cloud.githubusercontent.com/assets/1899308/18416557/28ae7914-7818-11e6-9992-07bfa32272d3.png">

I think the elements of each app have to be rearranged for a proper overall look. Maybe we have to take the design of the app store into account as well: https://apps.nextcloud.com/

jancborchardt commented:
@eppfel looks awesome! (Did I ask you already if you’re going to come to the [Nextcloud conference](https://conf.nextcloud.com)?) ;)

eppfel commented:
@jancborchardt You did and I will 😉

jancborchardt commented:
@eppfel @rullzer is a first version of this ready for getting in, or where do you need help? We should merge it to master soon so we can see how it works and polish it for Nextcloud 11. :)

MorrisJobke commented on settings/css/settings.css:
```diff
@@ -383,12 +383,24 @@ span.version {
+    margin: 0;
+    border-color: #eee;
+    border-width: 0 1px 1px 0;
+    border-style: solid;
```
Tabs vs spaces ;)

eppfel replied:
sry, fixed

eppfel commented:
@jancborchardt It needs to be fixed, because right now it affects the personal settings page as well. Besides that, the flexing seems robust, but the current version still has borders and I would like to get another opinion on that besides yours ;) Maybe there is time during the meetup, but otherwise the week is packed tight.

MorrisJobke commented:
@eppfel Ready to review? Then change the label ;)

eppfel commented:
@MorrisJobke Not, yet. I just fixed the spaces with an evil force push. But pssst :speak_no_evil: don't tell anyone

jancborchardt commented:
@MorrisJobke do you have any take on grid lines vs no grid lines? As @eppfel asked above ;)

MorrisJobke commented:
> @MorrisJobke do you have any take on grid lines vs no grid lines? As @eppfel asked above ;)

I would either go with a higher padding of 40px and no grid lines or leave the grid lines and 30px padding. The higher padding is needed to compensate for the missing visual guidelines. Generally speaking I like the no-gridlines version more, because we only have the line for the sidebar and horizontal lines in the files app for example, but never used vertical grid lines.

jancborchardt commented:
@eppfel you wanna try out the higher padding? Then let’s get this in and refine later. I’d hate to have this great improvement laying around for too long. ;)

eppfel commented:
Jep. I'll try that out. Was just a crazy last week, no time at all.

eppfel commented:
I only used the padding-right, but 50px, because this side has no sharp edge. Works for me, but I will open another issue for organizing the elements of each app.
I squashed my commits, so ready for gogo!

LukasReschke commented:
LGTM

MorrisJobke commented:
Tested and works 👍 

Tested in Firefox, Chrome, Safari, IE11 and Edge

jancborchardt commented:
Awesome work @eppfel! :) :rocket:
